### ISSUE TITLE
Use jakarta.xml instead of javax.xml

### ISSUE THREAD

**Issue by plebcity** [veraPDF/veraPDF-library#1314]
We are trying to upgrade our application to Spring Boot 3.0 but VeraPDF core is stopping us from doing that. It still uses the old javax.xml libraries instead of the new jakarta.xml.

Please change these 3 old dependencies to the new jakarta dependencies:

OLD
```
      <dependency>
        <groupId>javax.xml.bind</groupId>
        <artifactId>jaxb-api</artifactId>
        <version>2.4.0-b180830.0359</version>
      </dependency>

      <dependency>
        <groupId>com.sun.xml.bind</groupId>
        <artifactId>jaxb-impl</artifactId>
        <version>2.3.2</version>
      </dependency>

      <dependency>
        <groupId>com.sun.xml.bind</groupId>
        <artifactId>jaxb-core</artifactId>
        <version>2.3.0.1</version>
      </dependency>
```

NEW
```
      <dependency>
          <groupId>jakarta.xml.bind</groupId>
          <artifactId>jakarta.xml.bind-api</artifactId>
          <version>4.0.0</version>
      </dependency>

      <dependency>
        <groupId>org.glassfish.jaxb</groupId>
        <artifactId>jaxb-runtime</artifactId>
        <version>4.0.1</version>
      </dependency>

      <dependency>
        <groupId>org.glassfish.jaxb</groupId>
        <artifactId>jaxb-core</artifactId>
        <version>4.0.1</version>
      </dependency>
```

This is blocking for us and will be blocking for many other people trying to upgrade to latest Spring Boot or Wildfly

**Comment by bdoubrov** [comment]
veraPDF doesn't plan dropping Java8 support yet. And jakarta.xml 4.0 unfortunately requires Java 11. Would jakarta.xml 2.3 (the latest supported by Java 8) still work for you?

**Comment by plebcity** [comment]
> veraPDF doesn't plan dropping Java8 support yet. And jakarta.xml 4.0 unfortunately requires Java 11. Would jakarta.xml 2.3 (the latest supported by Java 8) still work for you?

We would need 3.0+

**Comment by bsanchezb** [comment]
Hello,

I join the request, as javax in a used dependency is a real blocker when upgrading to jakarta.
We need Jakarta 3.0 to support Java 8.

Would highly appreciate the upgrade if feasible.

Best regards,
Aleksandr.

**Comment by carlwilson** [comment]
Hi @plebcity & @bsanchezb, we've done a little investigation and will be looking to address this issue for the next release. To confirm, it's really just compatible Maven packages with the appropriate JAR artefacts available, possibly under a separated group ID or with adjusted names?

If that's an appropriate solution, or not, let us know here and we will let you know here when we have the first builds available for testing from our own Maven repo.

**Comment by blommish** [comment]
> Hello @carlwilson,
> 
> The best practice seems to have javax and jakarta versions of a project to be supported in parallel, until dropping the javax version support. So probably you may create a 2.x version of VeraPdf with Jakarta support, and continue 1.x with javax. Please also note that projects on javax are not compatible with jakarta projects and vice versa. The future is Jakarta for all projects, but support of javax for some time could be interesting so your users will not be forced to upgrade to jakarta as soon as possible.
> 
> Best regards, Aleksandr.

+1 And we also like this to be updated

**Comment by carlwilson** [comment]
Rest assured that we are actively looking at a solution and will post progress here. That's not supposed to be a ban on reminders. Good to know that there's some interest.

**Comment by gexclaude** [comment]
+1

**Comment by yagee-de** [comment]
+1

**Comment by bdoubrov** [comment]
@bsanchezb and others, as far as we can see, Spring Boot 3 uses jakarta 4 by default. We plan to release a special veraPDF artifact that will use jakatra 4 and will require Java 11+. Is this acceptable?

Or the configuration of veraPDF + jakarta 3 + Java 8 is also needed for some reasons?

**Comment by gexclaude** [comment]
Yes, Spring Boot 3.0 use jakarta.xml.bind:jakarta.xml.bind-api:4.0.0

In our case Java 11+ is fine, we use Java 17.

**Comment by plebcity** [comment]
Spring Boot 3 mandates Java 17.

**Comment by bdoubrov** [comment]
Thanks for the feedback! We aim to support Java 8 for the CLI and Desktop versions of veraPDF for a while, but we understand that most of veraPDF library integrators have already dropped Java 8

**Comment by gexclaude** [comment]
Just wanted to check - is there already a release estimate?

**Comment by gexclaude** [comment]
Thank you @bdoubrov . Just tested validation-model-jakarta:1.23.4 and our tests pass.

**Comment by blommish** [comment]
When do you think this will this be released?

**Comment by MaximPlusov** [comment]
Included into release 1.24
